docs/guide/extending/extending_thumbnails.md

```diff
@@ -0,0 +1,128 @@
+# Extending thumbnails
+
+Thumbnails enable you to easily choose an image for a specific Content Type.
```

Comment:
This introduction sentence could be improved, I think.
- I'd use the terms "the thumbnails API", as it is what we provide.
- The process is not "content type" specific. The description below does say that it iterates through fields definitions (independently of content type), and then falls back to the content type's icon.